src/ixdat/readers/zilien.py

```diff
@@ -162,150 +189,321 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
             "series_list": series,
             "aliases": aliases,
-            "tstamp": timestamp,
-            "metadata": metadata,
+            "tstamp": self._timestamp,
+            "metadata": self._metadata,
         }
         measurement_kwargs.update(kwargs)
         self._measurement = cls(**measurement_kwargs)
         return self._measurement
 
-    def _form_series(self, data, metadata, timestamp, series_headers, column_headers):
-        """Form the series and series aliases
+    @staticmethod
+    def _read_metadata(file_handle):
+        """Read metadata from `file_handle`"""
 
-        Args:
-            data (numpy.array): The data block of the tsv file as an array
-            metadata (dict): Extracted metadata
-            timestamp (float): The timestamp of the measurement
-            series_headers (List[str]): List of series headers, slots with empty strings
-                means the same as the last non-empty one
-            column_headers (List[str]): List of column headers
+        # The first 4 lines always include the file version, number of header lines,
+        # number of data header lines and data start line in this order.
+        # Backwards compatibility is ensured, because the one extra line read will be
+        # just added to the 'metadata' dict and ignored later.
+        metadata = {}
+        fixed_metadata_lines_amount = 4
+        for _ in range(fixed_metadata_lines_amount):
+            key, value = parse_metadata_line(file_handle.readline())
+            metadata[key] = value
+
+        # read the rest when the total amount is known
+        for _ in range(metadata["num_header_lines"] - fixed_metadata_lines_amount):
+            key, value = parse_metadata_line(file_handle.readline())
+            metadata[key] = value
+
+        # version 1 of the file format is sometimes missing this value
+        if "file_format_version" not in metadata:
+            metadata["file_format_version"] = 1
+
+        series_headers = file_handle.readline().strip("\n").split("\t")
+        column_headers = file_handle.readline().strip("\n").split("\t")
+
+        return metadata, series_headers, column_headers
+
+    def _form_series(self):
+        """Form the series and series aliases
 
         Returns:
             List[Series], DefaultDict(str, List[str]): List of series and dict of aliases
         """
-        series = []
-        last_time_series = None
         aliases = defaultdict(list)
-        last_series_header = ""
+        series = []
 
-        # Iterate over numbered series and column headers
-        for column_number, (series_header, column_header) in enumerate(
-            zip(series_headers, column_headers)
-        ):
-            last_series_header = series_header or last_series_header
+        # Get non-empty series headers and their indices
+        # in order to process whole series chunks
+        series_cut_indices, nonempty_headers = self._get_series_splits(
```

Comment:
Would you mind refactoring this to `series_split_indeces`? As an unwritten (as of now) convention, "cut" in ixdat so far always refers to selecting in time (and "snip" refers to selecting in a spectrum's independent variable like wavelength or m/z). So that was confusing me here - I conldn't see how `series_cut_indices` was cutting anything with respect to time.